### ISSUE TITLE
Failed to open memory buffer (/dev/mem): Permission denied

### ISSUE THREAD

**Issue by RMPR** [nima/python-dmidecode#15]
The command:
```
python dmidecode.py
```
produces
```
** COLLECTED WARNINGS **
Failed to open memory buffer (/dev/mem): Permission denied
No SMBIOS nor DMI entry point found, sorry.
** END OF WARNINGS **
```
While
```
sudo dmidecode
```
displays correctly the information needed.
After research, I found out that the problem can be with python3-dmidecode because the following code:
```
import dmidecode
print(dmidecode.bios())
```
prints an empty dictionary.

**Comment by cz172638** [comment]
additional notes to this - strace dmidecode reveals, that it reads from sysfs:

openat(AT_FDCWD, "/sys/firmware/dmi/tables/smbios_entry_point", O_RDONLY) = 3
fstat(3, {st_mode=S_IFREG|0400, st_size=24, ...}) = 0
lseek(3, 0, SEEK_SET)                   = 0
read(3, "_SM3_V\30\3\0\0\1\0\327\30\0\0\0\0\16\0\0\0\0\0", 24) = 24
close(3)                                = 0
write(1, "Getting SMBIOS data from sysfs.\n", 32) = 32
write(1, "SMBIOS 3.0.0 present.\n", 22) = 22
write(1, "Table at 0x000E0000.\n", 21)  = 21
write(1, "\n", 1)                       = 1
openat(AT_FDCWD, "/sys/firmware/dmi/tables/DMI", O_RDONLY) = 3
fstat(3, {st_mode=S_IFREG|0400, st_size=6359, ...}) = 0
lseek(3, 0, SEEK_SET)                   = 0

while efi.c uses /sys/firmware/efi/systab only for address and dump itself is from memory which is not present.

best would be to get it from /sys/firmware/dmi like dmidecode does

**Comment by winndows** [comment]
meet similar issue on debian 11 when booting form EFI:

```
root@test:~/test# python3
Python 3.9.2 (default, Feb 28 2021, 17:03:44)
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import dmidecode; print(dmidecode.get_warnings())
/dev/mem (mmap): Operation not permitted
No SMBIOS nor DMI entry point found, sorry.

>>> quit()
root@test:~/test# dmidecode
# dmidecode 3.3
Getting SMBIOS data from sysfs.
SMBIOS 2.7 present.
161 structures occupying 10032 bytes.
Table at 0x0FF7301F.

Handle 0x0000, DMI type 0, 26 bytes
BIOS Information
        Vendor: VMware, Inc.
```

**Comment by oilian** [comment]
Hi,

is there any progress on this bug?
Is there any plan to fix python3-dmidecode to read from sysfs?

Thanks
Oliver

**Comment by lian-bo** [comment]
> Hi,
> 
> is there any progress on this bug? Is there any plan to fix python3-dmidecode to read from sysfs?
> 
> Thanks Oliver

Hi, Oliver
Thank you for the attention.
This issue has been fixed in the smbios-3.3.0 branch. Could you please have a try? If there are any issues, welcome to report it. Thanks.
